### PR TITLE
Add option to install language dictionaries

### LINKED ISSUE #535: Spell check automatically set to English
## My Setup
- Operating System: Ubuntu 14.04 LTS or Windows 7
- App Version: rocketchat_2.9.0_amd64.deb or rocketchat-setup-2.9.0.exe
- Server: 0.58.2

client side:
```
~ $ echo $LANG
de_DE.UTF-8
``` 

- [x] I have tested with the latest version
- [x] I can simulate the issue easily

## Description

spell check is automatically set to English. My RC Profile is set to Geman. In Firefox the spell check works very well on German, but in my Electron client it's English so everything I type is red underlined.

### Current Behavior

spell check is automatically set to English.

### Expected Behavior

spell check should be the language in RC Profile or disabled.

### PR DESCRIPTION
**Pull request by damon-myers**
@RocketChat/DesktopApp 

Closes #535 

This PR adds an option to the Spellchecker menu to install a new language dictionary. It works by copying the dictionary to the app's dictionary directory and adding it to the list of available dictionaries.

### CODE REVIEW

**Comment by gdelavald** [comment]
@damon-myers Hey! This is awesome, thanks for you contribution.
We'll review the PR and add comments if there are any changes needed.

File: src/public/lib/SpellCheck.js [hunk 1]
```diff
@@ -70,7 +86,7 @@ class SpellCheck {
      * dictionary that is valid based on system (defaults to en_US)
      */
     setEnabledDictionaries () {
-        const dictionaries = this.dictionaries;
+        const dictionaries = this.availableDictionaries;
```
Comment (alexbrazier) [inline src/public/lib/SpellCheck.js]: When testing this it seems to have enabled all dictionaries for me. I think it is caused by the change you made here.

**Comment by damon-myers** [comment]
@alexbrazier Thanks for catching that!

I corrected the dictionary issue you mentioned, and also tested the changes on Windows. I had only tested them on Linux beforehand, so I also made it so that the install menu option is only shown if you are not using system language dictionaries. (In that case you'd simply install the language via your operating system.)

**Comment by gdelavald** [comment]
Hey @damon-myers giving this branch a try, can you provide the dictionary you used to test this? Just checking if we can add this to the current release or if it's going to be postponed into the next version.
Thanks.

**Comment by damon-myers** [comment]
@gdelavald Sure thing, I used the dictionary .dic and .aff from here: https://cgit.freedesktop.org/libreoffice/dictionaries/plain/cs_CZ

I used Google to get some Czech text and it seemed to correct that correctly.

**Comment by monspid** [comment]
any news? it would be a huge improvment

**Comment by gdelavald** [comment]
@damon-myers Sorry for the delay on this, we have had some issues testing this branch, on osx and ubuntu we had some exceptions thrown when starting the app.
Could you check the functions that read from directories, since they seem to be causing errors in some cases?

**Comment by monspid** [comment]
@gdelavald anything new? do you have any logs with errors?

**Comment by damon-myers** [comment]
@monspid, Holy crap! Sorry! I got busy and didn't notice there were issues with this. I can take a look at the crashes this weekend and try to get the branch up to date.

**Comment by monspid** [comment]
It's a great NEWS :-) It would be a huge improvement

**Comment by damon-myers** [comment]
I rebased the branch against develop and made sure everything still works.

@gdelavald, I'm not able to reproduce the issue on Linux or Windows (Windows uses system dictionaries, so this code isn't really used there.) Do you have any more information on the exceptions being thrown?

**Comment by ThomasVuillaume** [comment]
Can't wait to see that coming in, I had to copy over es_ES my fr_FR for now

**Comment by gabrielloberg** [comment]
We would love to have this functionality.

**Comment by engelgabriel** [comment]
@damon-myers can you please resolve the conflicts so I can merge this?

**Comment by damon-myers** [comment]
@engelgabriel, of course! On it!

**Comment by NatoBoram** [comment]
It's been a while since I tried to use different languages in Rocket Chat. There's this [PR](https://github.com/atom/node-spellchecker/pull/69) from one year ago that *could* fix that but wasn't merged.

**Pull request merged**